src/rust/src/orchard_ffi/incremental_sinsemilla_tree_ffi.rs

```diff
@@ -128,9 +130,26 @@ pub extern "C" fn orchard_merkle_frontier_root(
 pub extern "C" fn orchard_merkle_frontier_num_leaves(
     tree: *mut bridgetree::Frontier<OrchardIncrementalTreeDigest, MERKLE_DEPTH>,
 ) -> usize {
-    unsafe { tree.as_ref() }
-        .and_then(|t| t.position())
-        .map_or(0, |p| <usize>::from(p) + 1)
+    let tree = unsafe {
+        tree.as_ref()
+            .expect("Orchard note commitment tree pointer may not be null.")
+    };
+
+    tree.position().map_or(0, |p| <usize>::from(p) + 1)
+}
+
+
+#[no_mangle]
+pub extern "C" fn orchard_merkle_frontier_dynamic_mem_usage(
+    tree: *mut bridgetree::Frontier<OrchardIncrementalTreeDigest, MERKLE_DEPTH>,
+) -> usize {
+    let tree = unsafe {
+        tree.as_ref()
+            .expect("Orchard note commitment tree pointer may not be null.")
+    };
+
+    size_of::<Box<bridgetree::Frontier<OrchardIncrementalTreeDigest, MERKLE_DEPTH>>>()
```

Comment:
This is going to return the size of the box (a pointer), rather than the size of the memory it has allocated:
```suggestion
    size_of::<bridgetree::Frontier<OrchardIncrementalTreeDigest, MERKLE_DEPTH>>()
```
However, I [made some tests](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=5b317117bc76edd525f4f5e160f5b9bc), and it looks like since we take this as a reference (and thus its type is `&bridgetree::Frontier<_>`, we could also do:
```suggestion
    size_of_val(tree)
```